graphql/resolvers.go

```diff
@@ -2846,6 +2846,24 @@ func (r *queryResolver) MainlineCommits(ctx context.Context, options MainlineCom
 	})
 	var mainlineCommits MainlineCommits
 	activatedVersionCount := 0
+
+	mostRecentVersion, err := model.GetMostRecentMainlineCommit(projectId)
+
+	if err != nil {
+		// This shouldn't realy happen, but if it does, we should return an error and log it
+		grip.Warning(message.WrapError(err, message.Fields{
+			"message":   "Error getting most recent version",
+			"projectId": projectId,
+		}))
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error getting most recent mainline commit: %s", err.Error()))
+	}
+
+	// If a commit is the most recent version, we return nil for PrevPageOrderNumber since there aren't any other newer versions
+	if mostRecentVersion.RevisionOrderNumber != versions[0].RevisionOrderNumber {
+		// PrevPageOrderNumber is the order number of the first commit on the screen + the amount of commits on the previous page
```

Comment:
I actually don't really understand _why_ this number is this.  In your description you mention that limit is fickle; why can we use limit to calculate this?